File: src/main/java/AppPVSchema.java
import com.voop.data.cleaning.logic.mars.mobile.page.MobilePageProtos.MobilePage;
import org.apache.flink.streaming.util.serialization.AbstractDeserializationSchema;
import java.io.IOException;


public class AppPVSchema extends AbstractDeserializationSchema<MobilePage> {

    public MobilePage deserialize(byte[] message) throws IOException {
        MobilePage pv = MobilePage.parseFrom(message);
        return pv;
    }

    public boolean isEndOfStream(MobilePage nextElement) {
        return false;
    }
}

File: src/main/scala/AppPVSchema.scala
import com.voop.data.cleaning.logic.mars.mobile.page.MobilePageProtos.MobilePage
import org.apache.flink.streaming.util.serialization.AbstractDeserializationSchema

/**
  * Created by whjiang on 2016/12/22.
  */
class AppPVSchema extends AbstractDeserializationSchema[MobilePage]{
  override def deserialize(bytes: Array[Byte]): MobilePage = {
    MobilePage.parseFrom(bytes)
  }
}
